mage_ai/frontend/pages/datasets/[slug].tsx

```diff
@@ -190,14 +244,119 @@ function Data() {
 
   const headEl = (
     <FlexContainer alignItems="justify-right" flexDirection="row-reverse" >
-      <Button 
-        onClick={viewColumns}
-      >
+      <Button onClick={viewColumns}>
         <Text bold> Column view </Text>
       </Button>
     </FlexContainer>
   );
 
+  type ActionProps = {
+    idx: number;
+    link?: () => void;
+    name: string;
+    numFeatures: number;
+    onClose: () => void;
+    showIdx?: boolean;
+  }
+
+  const Action = ({
+    idx,
+    link,
+    name,
+    numFeatures,
+    onClose,
+    showIdx,
+  }: ActionProps) => (
+    <RowCard
+      columnFlexNumbers={[0.5, 0.5, 12]}
+    >
+      {link &&
+        <Link
+          bold
+          noHoverUnderline
+          onClick={link}
+        >
+          Apply
+        </Link>
+      }
+      {showIdx && <Text>{idx+1}</Text>}
+      <FlexContainer>
+        <Text>{name},</Text>
+        <Spacing mr={1} />
+        <Text secondary>{pluralize("feature", numFeatures)}</Text>
+      </FlexContainer>
+      <FlexContainer>
+        {/* TODO: add View Code & Preview here */}
+        <Button
+          basic
+          iconOnly
+          onClick={onClose}
+          padding="0px"
+          transparent
+        >
+          <Close muted />
+        </Button>
+      </FlexContainer>
+    </RowCard>
+  );
+
+  const actionsEl = (
```

Comment:
Prob wudve been better to create a new component in the `components` directory called `SuggestionRow` or something similar instead of creating the component directly in the pages file, but for sake of time, we'll leave it here and reorganize the components later.

Reply:
Completely agree. Like we talked about earlier we'll have to make a pass through the frontend and componentize elements into separate files